### PR TITLE
back button change 

### PR DESCRIPTION
**Pull request by ejohnson44**
changed the back button to back out of the app when in the connection fragment or when the menu is open, and to open the menu otherwise

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/CoreActivity.java [hunk 1]
```diff
@@ -94,9 +98,43 @@ public void onCreate(Bundle savedInstanceState){
...
+    }
+    
+    @Override
+    public void onBackPressed() {
+        //if we are in the connect fragment or the menu is open, 
+        //back out of the app by pulling up the home screen
+        if(getTitle().equals(getString(R.string.select)) || getSlidingMenu().isMenuShowing()){
+            Intent startMain = new Intent(Intent.ACTION_MAIN);
+            startMain.addCategory(Intent.CATEGORY_HOME);
+            startMain.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/CoreActivity.java:165]: I think we may want FLAG_ACTIVITY_CLEAR_TASK as well...https://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_CLEAR_TASK

Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/CoreActivity.java:165]: I just tried it and that made it ridiculously slow

Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/CoreActivity.java:165]: I don't actually remember what I was thinking with my previous comment.  Yea, that makes sense.  Just ignore me :)

**Comment by theJenix** [comment]
:thumbsup:

**Comment by dgreenhalgh** [comment]
:shipit:

**Pull request merged**